ip-range-matcher/src/com/dataiku/dip/plugins/iprangematcher/IPRangeMatcher.java

```diff
@@ -0,0 +1,278 @@
+package com.dataiku.dip.plugins.iprangematcher;
+
+import com.dataiku.dip.datalayer.Column;
+import com.dataiku.dip.datalayer.Processor;
+import com.dataiku.dip.datalayer.Row;
+import com.dataiku.dip.datalayer.SingleRowProcessor;
+
+import com.dataiku.dip.shaker.model.StepParams;
+import com.dataiku.dip.shaker.processors.Category;
+import com.dataiku.dip.shaker.processors.ProcessorMeta;
+import com.dataiku.dip.shaker.processors.ProcessorTag;
+import com.dataiku.dip.shaker.server.ProcessorDesc;
+import com.dataiku.dip.util.ParamDesc;
+
+import com.dataiku.dip.utils.ErrorContext;
+import com.google.common.collect.Sets;
+import org.apache.commons.lang.StringUtils;
+
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class IPRangeMatcher extends SingleRowProcessor implements Processor {
+    public static final Pattern RANGE_PATTERN = Pattern.compile("(\\d+)\\.(\\d+)\\.(\\d+)\\.(\\d+)[-_\\t ]{1,3}(\\d+)\\.(\\d+)\\.(\\d+)\\.(\\d+)");
+    public static final Pattern CIDR_PATTERN = Pattern.compile("(\\d+)(?:.(\\d)+)?(?:.(\\d)+)?(?:.(\\d)+)?\\/(\\d+)");
+    public static final Pattern IP_PATTERN = Pattern.compile("(\\d+)\\.(\\d+)\\.(\\d+)\\.(\\d+)");
+
+    private enum DetectorProcessorAction {
+        CREATE_FLAG_BOOLEAN,
+        CREATE_FLAG_MATCH,
+        DELETE_ROWS,
+        KEEP_ROWS,
+        CLEAR_CELLS
+    }
```

Comment:
Should reuse existing `FilterAndFlagProcessor` instead, all actions are already implemented, only have to implement cell matching.